tests/client/test_client.py

```diff
@@ -165,3 +176,60 @@ def test_get_missing(client, requests_mocker):
 
     # It should explain the problem
     assert "repo1 was not found" in str(error.value)
+
+
+def test_can_get_maintenance_report(client, requests_mocker):
+    maintenance_report = {
+        "last_updated": "2019-08-15T14:21:12Z",
+        "last_updated_by": "Content Delivery",
+        "repos": {
+            "repo1": {
+                "message": "Maintenance Mode Enabled",
+                "owner": "Content Delivery",
+                "started": "2019-08-15T14:21:12Z",
+            }
+        },
+    }
+    requests_mocker.get(
+        "https://pulp.example.com/pulp/isos/redhat-maintenance/repos.json",
+        json=maintenance_report,
+    )
+
+    report = client.get_maintenance_report().result()
+
+    assert requests_mocker.call_count == 1
+    assert isinstance(report, MaintenanceReport)
+    assert report.last_updated_by == "Content Delivery"
+    assert report.last_updated == "2019-08-15T14:21:12Z"
+    assert report.entries[0].message == "Maintenance Mode Enabled"
+
+
+def test_non_maintenance_report(client, requests_mocker):
+    requests_mocker.get(
+        "https://pulp.example.com/pulp/isos/redhat-maintenance/repos.json",
+        text="Not Found",
+        status_code=404,
+    )
+
+    report = client.get_maintenance_report().result()
+    assert report.last_updated_by is None
+    assert report.entries == ()
+
+
+def test_set_maintenance(client, requests_mocker):
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/repositories/search/",
+        [{"json": [{"id": "redhat-maintenance", "notes": {"_repo-type": "iso-repo"}}]}],
+    )
+
+    report = MaintenanceReport()
+
+    with patch("pubtools.pulplib.FileRepository.upload_file") as mocked_upload:
+        with patch("pubtools.pulplib.Repository.publish") as mocked_publish:
+            mocked_upload.return_value = f_return()
```

Comment:
Test isn't really thorough enough, the uploaded content is never tested.  Seems like the test would pass even if the code uploaded garbage every time?